torch/csrc/autograd/python_engine.cpp

```diff
@@ -178,7 +179,9 @@ PyObject *THPEngine_run_backward(PyObject *self, PyObject *args, PyObject *kwarg
       "backward() called inside torch.vmap. This is not supported, "
       "please call backward() outside torch.vmap or instead use "
       "torch.autograd.grad inside torch.vmap");
-
+  if (backward_api_called && create_graph) {
```

Comment:
Why not move this in the vanilla engine execute so that you have a single place where you need to have this code?

Reply:
That's indeed a better place to have this